### PR TITLE
Cleans up all Rubocop offenses. Syncs diffs between GitHub and Heroku…

### PR DESCRIPTION
**Pull request by jakedjohnson**
… repos

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,6 +7,7 @@ gem 'rails', '5.0.0.1'
 gem 'pg', '0.18.4'
 gem 'nokogiri', '1.6.8'
 gem 'rest-client'
+gem 'rubocop', require: false
```
Comment (bnd5k) [inline Gemfile]: We only need this in development mode. 

If you leave it here, this gem will be loaded when the app runs in production. That'll take up memory.  If you're not using the gem in production, then don't rob yourself of the memory.

File: app/controllers/websites_controller.rb [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
+                when 'mobile_asc'
+                  Website.mobile_asc
+                when 'mobile_desc'
+                  Website.mobile_desc
+                when 'desktop_asc'
+                  Website.desktop_asc
+                when 'desktop_desc'
+                  Website.desktop_desc
+                else
+                  Website.all
```
Comment (bnd5k) [inline app/controllers/websites_controller.rb]: The 'all' isn't going to organize the records in a way that's meaningful to a human.  I think it'll just order them by their db id.  Can you make it default to ordering `Website.mobile_asc` by default?

File: lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb [hunk 3]
```diff
@@ -3,41 +3,40 @@
...
+      rescue RestClient::BadRequest => e
+        Rails.logger.error(e)
+        return 0
+        # Website score columns placeholder, needed for scope querying
+      rescue RestClient::InternalServerError => e
+        Rails.logger.error(e)
+        return 0
       end
 
       def craft_api_call(site_url, strategy)
```
Comment (bnd5k) [inline lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb:34]: Maybe try something like this?  

```
base_url = 'https://www.googleapis.com/pagespeedonline/v1/runPagespeed'
params = "?url=#{site_url}&key=#{ENV['PAGESPEED_KEY']}&prettyprint=false&strategy=#{strategy}"
base_url + params
```

File: lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb [hunk 4]
```diff
@@ -19,29 +18,27 @@ def self.parse(html_string, xypn_url)
...
-          Rails.logger.error(e)
-          return "Business URL Not Found" 
-          # db placeholder, clean view rendering. Evaluation handles the API error.
-        end
+      def parse_business_site
+        @nokogiri_object.xpath('//p[@class="advisor-website"]'\
+          '/a/@href').first.value
+      rescue StandardError => e
+        Rails.logger.error(e)
+        return 'Business URL Not Found'
```
Comment (bnd5k) [inline lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: Do you ever evaluate this string?  To put it another way: do you care that it returns a string?  You could just do `return true` or `return`.

File: app/views/websites/index.html.erb [hunk 5]
```diff
@@ -5,6 +5,17 @@
...
+    <div class="col-sm-6 col-sm-offset-3 text-center">
+      <p>
+        The desktop and mobile scores listed are calculated by Google's PageSpeed Insights 
+        and represent a website's overall speed and performance. PageSpeed Insights analyzes 
+        a page to see if it follows Google's recommendations for making a page render in under a 
+        second. Research has shown that any delay longer than a second will cause the user to 
+        interrupt their flow of thought, creating a poor experience. A higher score is better 
+        and a score of 85 or above indicates that the page is performing well.
+      </p>
+    </div>
```
Comment (bnd5k) [inline app/views/websites/index.html.erb:26]: Last minute addition: can you add this link? 

https://www.doubleclickbygoogle.com/articles/mobile-speed-matters/

File: lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb (hunk 4 above)
Comment (jakedjohnson) [reply lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: `return true` put a string 't' in the db column which looked icky pirnted on the index page, so I went with the empty return/no string and the API call doesn't throw an error or anything wonky

File: lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb (hunk 3 above)
Comment (jakedjohnson) [reply lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb:34]: Did the refactor and Rubocop is still grumpy about the params string being too long, so still ended up having to use an escape character. I like the variables though, very verbose.

File: lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb (hunk 4 above)
Comment (bnd5k) [reply lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: Hmm...  This code should not be saving anything to the DB there's been any sort of failure.

Let's just return 'nil' for now.

File: lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb [hunk 6]
```diff
@@ -19,29 +18,26 @@ def self.parse(html_string, xypn_url)
...
-          @nokogiri_object.xpath('//p[@class="advisor-website"]/a/@href').first.value
-        rescue Exception => e
-          Rails.logger.error(e)
-          return "Business URL Not Found" 
-          # db placeholder, clean view rendering. Evaluation handles the API error.
-        end
+      def parse_business_site
+        @nokogiri_object.xpath('//p[@class="advisor-website"]'\
+          '/a/@href').first.value
+      rescue StandardError => e
```
Comment (bnd5k) [inline lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: StandardError is pretty broad.  In general, you want to rescue specific errors.  What error does nokogiri throw here when things go awry?

Whatever error nokogiri is throwning is probably subclass of StandardError, which is why this works.

When you're writing your own gems and apps, you'll find it helpful to raise very specific errors and you can define custom error classes like this.

`class MySuperSpecificError < StandardError ; end`

the semicolon bit is just shorthand for making your error class fit only 1 line.

But the idea is that you'll raise very specific errors to alert consumers what when awry.

File: lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb (hunk 3 above)
Comment (bnd5k) [reply lib/xypn_seo/seo_evaluation/advisor_site_evaluation.rb:34]: Hmm...I'd be OK with violating the rubocop's rules here.  They're not gospel.  They guidance.

Alternatively, does this approach work?  It's a little more robust than just string interpolation.

```
params = {
  url: #{site_url},
  key: #{ENV['PAGESPEED_KEY']},
  prettyprint: false,
  strategy: #{strategy}
}
uri = URI::HTTP.build(host: "https://www.googleapis.com/pagespeedonline/v1/runPagespeed", query: params.to_query)

Play around with this if you have time, if not, ship what you've got.  No sense is spending a ton of time on this issue.  It's very minor.
```

File: lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb (hunk 4 above)
Comment (jakedjohnson) [reply lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: I checked in the DB, and as it stands the current function is leaving the space 'nil', so we're good there.

Comment (bnd5k) [reply lib/xypn_seo/xypn_scraper/scrape_advisor_profile_page.rb]: 👍

**Review by bnd5k — APPROVED** [review]

**Comment by bnd5k** [comment]
👍

**Pull request merged**